### PR TITLE
some pep8 fixes in coding, in particular E275

### PR DESCRIPTION
**Pull request by fchapoton**
this fixes various pycodestyle warnings in coding, in particular E275

`E275 missing whitespace after keyword`

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.

### CODE REVIEW

File: src/sage/coding/cyclic_code.py [hunk 1]
```diff
@@ -577,7 +577,7 @@ def field_embedding(self):
               To:   Finite Field in z3 of size 2^3
               Defn: 1 |--> 1
         """
-        if not(hasattr(self, "_field_embedding")):
+        if not (hasattr(self, "_field_embedding")):
```
Comment (dcoudert) [inline src/sage/coding/cyclic_code.py]: remove parenthesis around `hasattr(...)`

File: src/sage/coding/grs_code.py [hunk 2]
```diff
@@ -1235,8 +1235,8 @@ def _decode_to_code_and_message(self, r):
         l0 = n-1-t
         l1 = n-1-t-(k-1)
         S = matrix(C.base_field(), n, l0+l1+2,
-                    lambda i, j: (C.evaluation_points()[i])**j if j < (l0+1)
-                    else r_list[i]*(C.evaluation_points()[i])**(j-(l0+1)))
+                   lambda i, j: (C.evaluation_points()[i])**j if j < (l0+1)
+                   else r_list[i]*(C.evaluation_points()[i])**(j-(l0+1)))
```
Comment (dcoudert) [inline src/sage/coding/grs_code.py]: the alignment of the `else` statement is not consistent with parenthesis

File: src/sage/coding/grs_code.py [hunk 3]
```diff
@@ -1574,7 +1574,7 @@ def _partial_xgcd(self, a, b, PolRing):
 
         r = b
         prev_r = a
-        while(r.degree() >= stop):
+        while (r.degree() >= stop):
```
Comment (dcoudert) [inline src/sage/coding/grs_code.py]: you could remove the parenthesis

File: src/sage/coding/grs_code.py [hunk 4]
```diff
@@ -2165,7 +2165,7 @@ def _partial_xgcd(self, a, b, PolRing):
         prev_r = a
         r = b
 
-        while(r.degree() >= t.degree()):
+        while (r.degree() >= t.degree()):
```
Comment (dcoudert) [inline src/sage/coding/grs_code.py]: remove parenthesis

File: src/sage/coding/linear_code.py [hunk 5]
```diff
@@ -877,7 +877,7 @@ def chinen_polynomial(self):
             P = RT(C.zeta_polynomial())*T**(d - dperp)
             if is_even(n):
                 Pd = q**(k-n/2)*RT(Cd.zeta_polynomial())
-            if not(is_even(n)):
+            if not (is_even(n)):
```
Comment (dcoudert) [inline src/sage/coding/linear_code.py]: `not is_even(n)`

File: src/sage/coding/linear_code.py [hunk 6]
```diff
@@ -886,7 +886,7 @@ def chinen_polynomial(self):
             P = RT(C.zeta_polynomial())
             if is_even(n):
                 Pd = q**(k-n/2)*RT(Cd.zeta_polynomial())
-            if not(is_even(n)):
+            if not (is_even(n)):
```
Comment (dcoudert) [inline src/sage/coding/linear_code.py]: remove useless parenthesis

File: src/sage/coding/linear_code.py [hunk 7]
```diff
@@ -1660,13 +1660,13 @@ def permutation_automorphism_group(self, algorithm="partition"):
...
                     A = libgap(matCwt).MatrixAutomorphisms()
-                    Gp = A.Intersection2(Gp)  #  bottleneck 3
+                    Gp = A.Intersection2(Gp)  # bottleneck 3
                     if Gp.Size() == 1:
                         return PermutationGroup([()])
                     gens = Gp.GeneratorsOfGroup()
                     stop = 1                    # get ready to stop
                     for x in gens:              # if one of these gens is not an auto then don't stop
-                        if not(self.is_permutation_automorphism(Sn_sage(x))):
+                        if not (self.is_permutation_automorphism(Sn_sage(x))):
```
Comment (dcoudert) [inline src/sage/coding/linear_code.py]: remove useless parenthesis

File: src/sage/coding/linear_code.py [hunk 8]
```diff
@@ -1877,7 +1877,7 @@ def weight_distribution(self, algorithm=None):
             from sage.coding.binary_code import weight_dist
             return weight_dist(self.generator_matrix())
         elif algorithm == "leon":
-            if not(F.order() in [2,3,5,7]):
+            if not (F.order() in [2,3,5,7]):
```
Comment (dcoudert) [inline src/sage/coding/linear_code.py]: `if F.order() not in [2, 3, 5, 7]:`

File: src/sage/coding/linear_code_no_metric.py [hunk 9]
```diff
@@ -892,7 +892,7 @@ def is_subcode(self, other):
         """
         G = self.generator_matrix()
         for r in G.rows():
-            if not(r in other):
+            if not (r in other):
```
Comment (dcoudert) [inline src/sage/coding/linear_code_no_metric.py]: `if r not in other:`

File: src/sage/coding/punctured_code.py [hunk 10]
```diff
@@ -328,7 +328,7 @@ def structured_representation(self):
         C = self.original_code()
         pts = copy(self.punctured_positions())
         list_pts = list(pts)
-        while(isinstance(C, PuncturedCode)):
+        while (isinstance(C, PuncturedCode)):
```
Comment (dcoudert) [inline src/sage/coding/punctured_code.py]: useless parenthesis

File: src/sage/coding/reed_muller_code.py [hunk 11]
```diff
@@ -258,9 +258,9 @@ def __init__(self, base_field, order, num_of_var):
         # input sanitization
         if base_field not in FiniteFields():
             raise ValueError("the input `base_field` must be a FiniteField")
-        if not(isinstance(order, (Integer, int))):
+        if not (isinstance(order, (Integer, int))):
```
Comment (dcoudert) [inline src/sage/coding/reed_muller_code.py]: useless parenthesis

**Review by dcoudert — CHANGES_REQUESTED** [review]

**Review by dcoudert — APPROVED** [review]
LGTM.

**Comment by fchapoton** [comment]
merci pour la relecture attentive !

**Pull request merged**